runtime/reconcilers/source.go

```diff
@@ -392,10 +393,27 @@ func (r *SourceReconciler) ingestSource(ctx context.Context, src *runtimev1.Sour
 	if ingestionLimit != nil {
 		opts.LimitInBytes = *ingestionLimit
 	}
+	transferStart := time.Now()
 	err = t.Transfer(ctx, srcConfig, sinkConfig, opts, drivers.NoOpProgress{})
 	if limitExceeded {
 		return drivers.ErrIngestionLimitExceeded
 	}
+
+	defer func() {
+		transferLatency := time.Since(transferStart).Milliseconds()
+		commonDims := []attribute.KeyValue{
+			attribute.String("source", srcConn.Driver()),
+			attribute.String("destination", sinkConn.Driver()),
+			attribute.Bool("cancelled", errors.Is(outErr, context.Canceled)),
+			attribute.Bool("failed", outErr != nil),
+			attribute.Bool("limit_exceeded", limitExceeded),
+			attribute.Int64("limit_bytes", opts.LimitInBytes),
+		}
+		r.C.Activity.Emit(ctx, "ingestion_ms", float64(transferLatency), commonDims...)
+
+		// TODO: emit the number of bytes ingested (this might be extracted from a progress)
+	}()
```

Comment:
The `if limitExceeded` is checked before the `defer`. It would be best to move the `defer` to right after `transferStart` is declared (before calling `Transfer`)